2017 Robot/src/org/usfirst/frc/team4999/lights/NeoPixels.java

```diff
@@ -8,19 +8,24 @@
 class NeoPixelsIO extends I2C {
 	
 	private final Packet displayPacket;
+	private final ByteBuffer buffer;
+	
+	private final int MAX_PACKET_SIZE = 16;
 	
 	public NeoPixelsIO(Port port, int deviceAddress) {
 		super(port, deviceAddress);
 		displayPacket = new Packet();
+		buffer = ByteBuffer.allocateDirect(MAX_PACKET_SIZE);
 	}
 	
 	public boolean writePacket(Packet packet) {
-		return writeBulk(packet.fillBuffer(), packet.getPacketSize());
+		packet.fillBuffer(buffer);
```

Comment:
Knowledge of the existence and necessity for the ByteBuffer lies solely within the NeoPixelsIO class.  There is no reason that the Packet class should ever know anything about the ByteBuffer.  Move the implementation of fillBuffer here.